### ISSUE TITLE
[Bug Report] ttnn.max wrong results 

### ISSUE THREAD

**Issue by rfurko-tt** [tenstorrent/tt-metal#12662]
**Describe the bug**
ttnn.max returns unexpected results

**To Reproduce**
```
with ttnn.manage_device(device_id=0) as device:
    x = -torch.ones((1, 2, 3, 4), dtype=torch.float32)
    x_tt = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
    y_tt = ttnn.max(x_tt, dim=-1)
    print(ttnn.to_torch(x_tt))
    print(ttnn.to_torch(y_tt))
```

Output:
```
TorchTensor([[[[-1., -1., -1., -1.],
               [-1., -1., -1., -1.],
               [-1., -1., -1., -1.]],

              [[-1., -1., -1., -1.],
               [-1., -1., -1., -1.],
               [-1., -1., -1., -1.]]]], dtype=torch.bfloat16)
TorchTensor([[[[0.],
               [0.],
               [0.]],

              [[0.],
               [0.],
               [0.]]]], dtype=torch.bfloat16)
```

**Expected behavior**
Resulting value should be -1. 

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- Version of software: a7d9204fc9cdf579e0a512e3c26fb291b398c75b

**Comment by dmakoviichuk-tt** [comment]
Hi @mywoodstock it blocks us from solving other issues for example unstable softmax: https://github.com/tenstorrent/tt-metal/issues/12661
It should be fixed ASAP.
cc @jvasilje

**Comment by mywoodstock** [comment]
@dmakoviichuk-tt Don't think I'm the right person for op --`max` would be under elt-wise?

**Comment by yugaoTT** [comment]
Hey @caixunshiren,
is this bug the same one you mentioned earlier (the reduce issue), which requires LLK changes?

**Comment by caixunshiren** [comment]
> Hey @caixunshiren,
> is this bug the same one you mentioned earlier (the reduce issue), which requires LLK changes?

I'm not sure if they are the exact same issue, but it is related to reduce_revert_delta function

here is the issue: https://github.com/tenstorrent/tt-metal/issues/9370

**Comment by amahmudTT** [comment]
@caixunshiren Does it fail on the main branch too now ? We put a workaround for the issue in #9370 which fixes it, but we have not closed the issue since it is a temporary workaround.

**Comment by caixunshiren** [comment]
> @caixunshiren Does it fail on the main branch too now ? We put a workaround for the issue in #9370 which fixes it, but we have not closed the issue since it is a temporary workaround.

No it doesn't! We haven't observed ndpcc so far

**Comment by yugaoTT** [comment]
I think the problem is the same as in softmax, the padded values are all 0s, and when use max/reduce it will pick those values, the Softmax fix is here: https://github.com/tenstorrent/tt-metal/pull/13068

@dmakoviichuk-tt please assign this ticket to the reduce op owner as I'm unfamiliar with it, I can walk through the details of the fix.

**Comment by amahmudTT** [comment]
So I had a look at ttnn/cpp/ttnn/operations/reduction/generic/device/kernels/compute/reduce_w.cpp
The function does not call reduce_revert_delta(), so is the ND PCC you are facing may be related to the reduce_revert_delta when you are running a big program, but as for the above test case it is not related to #9370

**Comment by amahmudTT** [comment]
Could you point me to the issue/code where you are seeing the ND issue ? @caixunshiren

**Comment by caixunshiren** [comment]
> Could you point me to the issue/code where you are seeing the ND issue ? @caixunshiren

Yes it is this issue: https://github.com/tenstorrent/tt-metal/issues/9370

**Comment by jvasilje** [comment]
@ttmtrajkovic @dmakoviichuk-tt @yugaoTT  - who is working on a fix for this issue? there are 3 of you assigned, and it's a P0 bug/=. 

A P0 bug has 1 assignee, and should receive daily updates on progress - otherwise it's not a P0.

**Comment by amahmudTT** [comment]
> @ttmtrajkovic @dmakoviichuk-tt @yugaoTT - who is working on a fix for this issue? there are 3 of you assigned, and it's a P0 bug/=.
> 
> A P0 bug has 1 assignee, and should receive daily updates on progress - otherwise it's not a P0.

Initially there was some idea that this issue was related to an issue we had been looking at in llk (ND issue) so Milos was assigned. I am removing Milos from the assigned list.

**Comment by dmakoviichuk-tt** [comment]
@jvasilje  @amahmudTT  do you have any updates on this issues?

**Comment by dmakoviichuk-tt** [comment]
@amahmudTT  please don't answer my questions in my comments :)

**Comment by amahmudTT** [comment]
> @amahmudTT please don't answer my questions in my comments :) I started thinking I have a split consciousness.

Damn, I never knew this was even possible lol

**Comment by bbradelTT** [comment]
I changed the shape to (1, 2, 32, 32) and it looks like -1 values are returned. Next steps: I will need to look at how the padding is set up.

**Comment by dmakoviichuk-tt** [comment]
@bbradelTT  does autoformat cause issue here too?

**Comment by bbradelTT** [comment]
In a way. autoformat uses pad, which ignores the pad value and either pads with 0s or fails with an assert (depending on the dimensions passed in). I opened up a P0 since that's blocking my other P0 as well: https://github.com/tenstorrent/tt-metal/issues/15511

**Comment by bbradelTT** [comment]
@prajaramanTT yes this is still an open issue.

**Comment by bbradelTT** [comment]
For the first dims, there is a subsequent PR related with other issues: https://github.com/tenstorrent/tt-metal/pull/16989
